### ISSUE TITLE
LuxCore Add-on freezes Blender on MacOS when switching to GPU Compute.

### ISSUE THREAD

**Issue by anshulpatria** [LuxCoreRender/BlendLuxCore#625]
Hi, Downwloaded the add-on for blender. 
It froze blender and got stuck. 
This is on MacOS, switch to GPU Compute, Click on rendered view. This happens even with the simplest geometry. Something I am doing wrong? 
![IMG-1150](https://user-images.githubusercontent.com/65661498/106245530-986b7c80-6232-11eb-92fa-2239ffa9d3ba.jpg)

**Comment by Theverat** [comment]
It might just be building the OpenCL kernels for the first time. You can check in more detail what's going on by starting Blender from a terminal and checking the messages there.
Since macOS OpenCL drivers are garbage, you might need a few hours for this to complete.

**Comment by jamespa16** [comment]
Yep, that's most likely what's happening. The longest I've seen is about thirty minutes on my 2018 i7 Mac mini.

**Comment by CodeFHD** [comment]
Closing this issue after a review with the following reasons/conclusions:
- Issue affected old Blender + LuxCore versions that are no longer supported
- Explanation was provided at the time (OCL kernel compilation)
- No follow-up in 4 years

If you feel this issue is still relevant in 2025, please re-test it with a recent Blender + LuxCore version, then ask to reopen and provide the following information:
- Operating System
- CPU + GPU model
- Blender and LuxCoreRender version used for re-test
- Detailed steps how to reproduce the bug from an empty scene,
- provide a specific .blend-file
